cpp/cmake/thirdparty/CUDF_GetDLPack.cmake

```diff
@@ -36,6 +36,6 @@ function(find_and_configure_dlpack VERSION)
     set(DLPACK_INCLUDE_DIR "${dlpack_SOURCE_DIR}/include" PARENT_SCOPE)
 endfunction()
 
-set(CUDF_MIN_VERSION_dlpack 0.3)
+set(CUDF_MIN_VERSION_dlpack 0.5)
```

Comment:
I could have sworn there's some header changes in 0.5 that require us updating a little bit of code.

We also need to update the conda recipes / environment files based on this change. The integration repo as well: https://github.com/rapidsai/integration/blob/branch-21.06/conda/recipes/versions.yaml#L55-L56

We also need to get dlpack conda packages updated in conda-forge. Working on the conda-forge updates...

Reply:
I figured there were changes necessary, but I tried a clean build and everything seemed fine. :shrug:

I'll take a closer look.

Reply:
Seems like we might have to update our code as `dlpack`-`0.5` seems to be breaking libcudf build:

```
[ 31%] Building CXX object CMakeFiles/cudf.dir/src/io/comp/brotli_dict.cpp.o
[ 32%] Building CXX object CMakeFiles/cudf.dir/src/io/comp/cpu_unbz2.cpp.o
[ 32%] Building CUDA object CMakeFiles/cudf.dir/src/io/comp/debrotli.cu.o
[ 32%] Building CUDA object CMakeFiles/cudf.dir/src/io/comp/gpuinflate.cu.o
[ 33%] Building CUDA object CMakeFiles/cudf.dir/src/io/comp/snap.cu.o
[ 33%] Building CXX object CMakeFiles/cudf.dir/src/io/comp/uncomp.cpp.o
[ 33%] Building CUDA object CMakeFiles/cudf.dir/src/io/comp/unsnap.cu.o
In file included from /nvme/0/pgali/cudf/cpp/include/cudf/utilities/type_dispatcher.hpp:22,
                 from /nvme/0/pgali/cudf/cpp/include/cudf/utilities/traits.hpp:20,
                 from /nvme/0/pgali/cudf/cpp/include/cudf/column/column_view.hpp:19,
                 from /nvme/0/pgali/cudf/cpp/include/cudf/column/column.hpp:18,
                 from /nvme/0/pgali/cudf/cpp/include/cudf/column/column_factories.hpp:18,
                 from /nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:16:
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp: In function 'std::unique_ptr<cudf::table> cudf::detail::from_dlpack(const DLManagedTensor*, rmm::cuda_stream_view, rmm::mr::device_memory_resource*)':
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:140:33: error: 'const struct DLTensor' has no member named 'ctx'
  140 |   CUDF_EXPECTS(kDLGPU == tensor.ctx.device_type || kDLCPU == tensor.ctx.device_type ||
      |                                 ^~~
/nvme/0/pgali/cudf/cpp/include/cudf/utilities/error.hpp:63:7: note: in definition of macro 'CUDF_EXPECTS'
   63 |   (!!(cond)) ? static_cast<void>(0)                                 \
      |       ^~~~
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:140:69: error: 'const struct DLTensor' has no member named 'ctx'
  140 |   CUDF_EXPECTS(kDLGPU == tensor.ctx.device_type || kDLCPU == tensor.ctx.device_type ||
      |                                                                     ^~~
/nvme/0/pgali/cudf/cpp/include/cudf/utilities/error.hpp:63:7: note: in definition of macro 'CUDF_EXPECTS'
   63 |   (!!(cond)) ? static_cast<void>(0)                                 \
      |       ^~~~
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:141:41: error: 'const struct DLTensor' has no member named 'ctx'
  141 |                  kDLCPUPinned == tensor.ctx.device_type,
      |                                         ^~~
/nvme/0/pgali/cudf/cpp/include/cudf/utilities/error.hpp:63:7: note: in definition of macro 'CUDF_EXPECTS'
   63 |   (!!(cond)) ? static_cast<void>(0)                                 \
      |       ^~~~
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:145:14: error: 'const struct DLTensor' has no member named 'ctx'
  145 |   if (tensor.ctx.device_type != kDLCPU) {
      |              ^~~
In file included from /nvme/0/pgali/cudf/cpp/include/cudf/utilities/type_dispatcher.hpp:22,
                 from /nvme/0/pgali/cudf/cpp/include/cudf/utilities/traits.hpp:20,
                 from /nvme/0/pgali/cudf/cpp/include/cudf/column/column_view.hpp:19,
                 from /nvme/0/pgali/cudf/cpp/include/cudf/column/column.hpp:18,
                 from /nvme/0/pgali/cudf/cpp/include/cudf/column/column_factories.hpp:18,
                 from /nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:16:
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:148:25: error: 'const struct DLTensor' has no member named 'ctx'
  148 |     CUDF_EXPECTS(tensor.ctx.device_id == device_id, "DLTensor device ID must be current device");
      |                         ^~~
/nvme/0/pgali/cudf/cpp/include/cudf/utilities/error.hpp:63:7: note: in definition of macro 'CUDF_EXPECTS'
   63 |   (!!(cond)) ? static_cast<void>(0)                                 \
      |       ^~~~
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp: In function 'DLManagedTensor* cudf::detail::to_dlpack(const cudf::table_view&, rmm::cuda_stream_view, rmm::mr::device_memory_resource*)':
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:237:34: error: 'struct DLTensor' has no member named 'ctx'
  237 |   CUDA_TRY(cudaGetDevice(&tensor.ctx.device_id));
      |                                  ^~~
/nvme/0/pgali/cudf/cpp/include/cudf/utilities/error.hpp:104:33: note: in definition of macro 'CUDA_TRY'
  104 |     cudaError_t const status = (call);                            \
      |                                 ^~~~
/nvme/0/pgali/cudf/cpp/src/interop/dlpack.cpp:238:10: error: 'struct DLTensor' has no member named 'ctx'
  238 |   tensor.ctx.device_type = kDLGPU;
      |          ^~~
[ 34%] Building CUDA object CMakeFiles/cudf.dir/src/io/csv/csv_gpu.cu.o
```